fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/AlterJobCommand.java

```diff
@@ -0,0 +1,141 @@
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.trees.plans.commands;
+
+import org.apache.doris.analysis.StmtType;
+import org.apache.doris.catalog.Env;
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.job.base.AbstractJob;
+import org.apache.doris.job.base.JobExecuteType;
+import org.apache.doris.job.common.JobStatus;
+import org.apache.doris.job.exception.JobException;
+import org.apache.doris.job.extensions.insert.streaming.StreamingInsertJob;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.nereids.trees.plans.visitor.PlanVisitor;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.StmtExecutor;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Map;
+
+/**
+ * alter job command.
+ */
+public class AlterJobCommand extends AlterCommand implements ForwardWithSync {
+    // exclude job name prefix, which is used by inner job
+    private static final String excludeJobNamePrefix = "inner_";
+    private String jobName;
+    private Map<String, String> properties;
+    private String sql;
+
+    public AlterJobCommand(String jobName, Map<String, String> properties, String sql) {
+        super(PlanType.ALTER_JOB_COMMAND);
+        this.jobName = jobName;
+        this.properties = properties;
+        this.sql = sql;
+    }
+
+    public String getJobName() {
+        return jobName;
+    }
+
+    @Override
+    public StmtType stmtType() {
+        return StmtType.ALTER;
+    }
+
+    @Override
+    public void doRun(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        validate();
+        AbstractJob job = analyzeAndBuildJobInfo(ctx);
+        ctx.getEnv().getJobManager().alterJob(job);
+    }
+
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitAlterJobCommand(this, context);
+    }
+
+    private AbstractJob analyzeAndBuildJobInfo(ConnectContext ctx) throws JobException {
+        AbstractJob job = Env.getCurrentEnv().getJobManager().getJobByName(jobName);
+        if (job instanceof StreamingInsertJob) {
+            StreamingInsertJob originJob = (StreamingInsertJob) job;
+            String updateSQL = StringUtils.isEmpty(sql) ? originJob.getExecuteSql() : sql;
+            Map<String, String> updateProps =
```

Comment:
Users have to provide all propeerties in alter command?

Reply:
Already fix